docs/contributing.rst

```diff
@@ -2,12 +2,701 @@
+from local in the GitHub repository (either in your fork or in the actual UXarray repo)
+and the :code:`main` branch, if not other, in the UXarray's GitHub repository. There
+might be some exceptions to this generic case, which can always be learned through
+reading or discussed with the maintainers and community.
+
+.. note::
+    While pull requests are supposed to be based on changes that are ready to be tested,
+    reviewed, and eventually be merged to repositories; there is an exception to this:
+    Draft Pull Requests. We encourage Draft PRs if you just want to just start a
+    conversation about your code that isn’t in any state to be judged, and get feedback
+    as well as some guidance from others. Please read this GitHub `blog
+    <https://github.blog/2019-02-14-introducing-draft-pull-requests/>`_ about draft PRs.
+
+Please refer to Github's `Creating a pull request
+<https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request>`_
+guide for the instructions. We are especially avoiding to detail these instructions
+here as there are manly referrals to GitHub's graphical user interface, which might be
+changed in the future.
+
+.. note::
+    Below are significant things about pull requests that can be very helpful throughout
+    the entire contribution process (i.e. review and merge) process when performed:
+
+3.7.1. Link PR to Issue
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Don't forget to link your PR to the issue you have been working on! This will help not
+only keep track of issues easier but also get them closed automatically once your PR is
+merged. You can refer to GitHub's `Linking a pull request to an issue
+<https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue>`_
+for this.
+
+3.7.2. Review your changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Before you finalize opening the actual PR, it is a good practice to review the changes
+that you’ve made. You should be able to review all of the changes that will go into
+this PR just before you press the `Create pull request` button.
+
+If there is any changes you want to make in the PR, you can delay creating the PR
+and push new commits, revert existing changes, etc. You can then create the PR.
+
+3.7.3. Common Elements of Most Pull Requests
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Despite the fact that pull requests can differ regarding their purposes (e.g.
+correction of a single typo in only one file could make a PR), most of the PRs may
+consist of code changes that should, most of the time, include some other elements
+with it. Having all of such elements addressed in a PR could make the review and
+merge process a lot easier. Assuming such code changes, these are what might
+accompany them:
+
+3.7.3.1. Unit tests
+~~~~~~~~~~~~~~~~~~~
+
+Virtually all new Uxarray code needs to include unit tests of it implemented.
```

Comment:
```suggestion
Virtually all new UXarray code needs to include unit tests of the functionality implemented.
```
This sentence was a little clunky, this is a potential way to word it instead

Reply:
Done , thanks!